src/main/java/slimeknights/tconstruct/tables/recipe/material/MaterialRecipe.java

```diff
@@ -0,0 +1,122 @@
+package slimeknights.tconstruct.tables.recipe.material;
+
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.library.MaterialRegistry;
+import slimeknights.tconstruct.library.materials.IMaterial;
+import slimeknights.tconstruct.library.materials.MaterialId;
+import slimeknights.tconstruct.tables.TinkerTables;
+
+public class MaterialRecipe implements IRecipe<IInventory> {
+  protected final ResourceLocation id;
+  protected final String group;
+  protected final Ingredient ingredient;
+  protected final MaterialId materialId;
+  protected final int value;
+  protected final int needed;
+
+  public MaterialRecipe(ResourceLocation id, String group, Ingredient ingredient, int value, int needed, MaterialId materialId) {
+    this.id = id;
+    this.group = group;
+    this.ingredient = ingredient;
+    this.materialId = materialId;
+    this.value = value;
+    this.needed = needed;
+  }
+
+  @Override
+  public IRecipeType<?> getType() {
+    return TinkerTables.materialRecipeType;
+  }
+
+  @Override
+  public ResourceLocation getId() {
+    return this.id;
+  }
+
+  @Override
+  public ItemStack getIcon() {
+    return new ItemStack(TinkerTables.partBuilder);
+  }
+
+  @Override
+  public IRecipeSerializer<?> getSerializer() {
+    return TinkerTables.materialRecipeSerializer.get();
+  }
+
+  /**
+   * Used to check if a recipe matches current crafting inventory
+   */
+  @Override
+  public boolean matches(IInventory inv, World worldIn) {
```

Comment:
May need to check multiple slots when used in the tool station, though we can rethink that later when needed.